src/deploy.ts

```diff
@@ -609,3 +660,15 @@ function formatAge(age: number): string {
   }
   return `at ${new Date(Date.now() - age).toLocaleString("en")}`;
 }
+
+async function sha512File(path: string): Promise<string> {
+  const hasher = createHash("sha512");
+  const stream = createReadStream(path);
+  const promise = new Promise<void>((resolve, reject) => {
+    hasher.once("finish", resolve);
+    hasher.once("error", reject);
+  });
+  stream.pipe(hasher);
+  await promise;
+  return (hasher.read() as Buffer).toString("base64");
```

Comment:
How important is it to use streaming here? Can we simplify like so (which is a pattern I use elsewhere):

```suggestion
  return createHash("sha512").update(await readFile(path)).digest("base64");
```

Reply:
Alternatively, I feel like this would be cleaner:

```ts
async function sha512File(path: string): Promise<string> {
  return new Promise((resolve, reject) => {
    const hash = createHash("sha512");
    createReadStream(path)
      .pipe(hash)
      .on("error", reject)
      .on("finish", () => resolve(hash.digest("hex")));
  });
}
```

Reply:
I think the main difference is that this is working in parallel, with up to 8 files being read at once. The most constrained place we're likely to be deploying projects is in CI. Github [allocates](https://docs.github.com/en/actions/using-github-hosted-runners/about-larger-runners/about-larger-runners#specifications-for-general-larger-runners) 4GB of memory per core, and we run at most two parallel hashers per core. We can optimize for files no larger than 50MB, so it should be no problem to read the whole file into memory.

Reply:
Both readFile and createReadStream should be parallelizable.

Reply:
Yes, but createReadStream would only keep a fixed window in memory at any given time.  readFile will read the whole file into memory. If we had more memory pressure that could be a problem, but it doesn't matter here.

Reply:
I’m cool with you using the alternative `createReadStream` version I posted above if you prefer it.

Reply:
Since their isn't much memory pressure. I'm happy with the readFile version, it's what I put in my latest commit. It's simple and small enough that I can inline it into the function instead of wanting to put it in another function, which is nice.